health/health_log.c

```diff
@@ -132,6 +134,7 @@ inline ssize_t health_alarm_log_read(RRDHOST *host, FILE *fp, const char *filena
 
     netdata_rwlock_rdlock(&host->health_log.alarm_log_rwlock);
 
+    netdata_rwlock_rdlock(&host->labels_rwlock);
```

Comment:
Why do you hold the labels lock around this loop? I can't find any access to host->labels inside the locked section.

Reply:
When I was developing, I was thinking in the possibility to do not load the logs case the labels were not present, but I changed my mind late and I forgot to remove this lock.